### ISSUE TITLE
preview:generate-all - only uses one php process?

### ISSUE THREAD

**Issue by LukeRepko** [rullzer/previewgenerator#74]
This may have been answered elsewhere, and I apologize in advance if it has, however I've not been able to find the answer after looking through open/closed issues for previewgenerator.

I have around 500,000 jpegs of all sizes that I am running, **preview:generate-all** to generate thumbs for. Obviously it's taking a while as expected, but I noticed almost immediately (two days ago) when I started this that **only one** php process has spawned since. I honestly can't figure out a good way to run multiple iterations of preview:generate-all safely. Consequently, thumb generation is extremely slow with only one core being used (Configuration issue on my end possibly, or just a limitation?).

If not a config issue in my NC 12 setup, is there anything that could be done to run multiple processes of this in parallel? I assume (maybe ignorantly) that one could run multiple processes of this if the, "scan-by-path" option was merged  #45 ? It seems important to utilize multiple threads for situations when new installations of Next Cloud are stood up, and we want to mass generate thumbs (or for bulk imports). 

Once a file gets its thumbs generated, is the file marked as such in the database so if the generation process does get hung up / crashes on a corrupt file, generate-all can be run again and will skip over files with existing thumbs? I'd test this now, but am already a couple days in and would rather not chance it. 

There should be a "--ignore-generated" arg where if thumbs exist, don't re-generate (if it's not already built-in / default behavior). 

Thanks for any help here!

**Comment by LukeRepko** [comment]
I merged #45 into my fork of previewgenerator, and found it working well to run against multiple sub directories at once in lieu of letting **preview:generate-all** run by itself against one directory/file at a time. 

I ended up stopping the in preview:generate-all of my 500,000 jpegs to merge that path option, and it's working great! I first tried to run multiple instances of preview:generate-all but that traced out pretty quick. 

I also found that files that have already had previews generated get scanned, and if thumbs are found, the app moves on. Since I found the answers to both of my concerns, I'm going to leave this open only because it would be nice to have preview:generate-all use more resources, either by default or with a flag. The only alternative seems to be having it run on a specified path. If you feel this is good enough, please feel free to close this.

**Comment by rullzer** [comment]
Yes it only uses 1 process. Threading in php is not so trivial.

**Comment by LukeRepko** [comment]
Thanks for the confirmation! :)

**Comment by rajil** [comment]
I am doing a './occ preview:generate-all' for the first time and the process is slow. Can i manually invoke multiple './occ preview:generate-all' without any untoward effect?
